OSDOCS - 18533 [ROSA]: FedRAMP Security Configuration Guide (#113973)
openshift/openshift-docs: modules/rosa-fedramp-accounts-config.adoc

// Module included in the following assemblies:
//
// * rosa_govcloud/rosa-fedramp-security-config.adoc

:_mod-docs-content-type: REFERENCE

[id="rosa-fedramp-accounts-config_{context}"]
= Red{nbsp}Hat {OCP} FedRAMP Rev5 secure configuration guidance for top-level administrative and privileged accounts

[role="_abstract"]
{product-title} gives you comprehensive frameworks to find roles, isolation boundaries, and operational functions for top-level administrative or privileged accounts. These frameworks apply within a Red{nbsp}Hat {OCP} Federal Risk and Authorization Management Program (FedRAMP) Rev5 Secure Configuration Guidance (SCG) environment.

The guidance you receive addresses the SCG-CSO-RSC (Cloud Service Offering - Recommended Secure Configuration) requirement, including both cluster-wide top-level administrative accounts and namespace-scoped privileged accounts. The parameters in both administrative and privilege accounts help you deploy, operate, decommission, and secure access among your configurations.

> For the most up-to-date account features and structural prerequisites, review the official Red{nbsp}Hat {OCP-short} documentation page for your active deployment service. Your browser must have JavaScript enabled to access and interact with the {OCP-short} documentation portal.

Reviewer: Maybe a link here and update this to the docs portal:
```suggestion
For the most up-to-date account features and structural prerequisites, review the official Red{nbsp}Hat {OCP-short} documentation page for your active deployment service. Your browser must have JavaScript enabled to access and interact with the Red{nbsp}Hat documentation portal.
```
or whatever the right term here is, i think we should check to make sure we're using the right terminology

> *IMPORTANT:*

Reviewer: suggestion to make this an admonition if it's important for customers to know:

```suggestion
[IMPORTANT]
====
```

> * To access the {cluster-manager-short} infrastructure for the first time, use the temporary, local `rosa create admin` account generated during the initial installation phase.

Reviewer: the `{cluster-manager-short}` attribute is not resolving

> * These root credentials reside securely in the `auth/kubeadmin-password ` file within your local installation directory.

Reviewer 2: The creds are piped to STDOUT when you do `rosa create admin`, not stored on disk

> * To comply with FedRAMP identity boundaries, you must use this account primarily to configure corporate identity infrastructure.

Reviewer: ```suggestion
* To comply with FedRAMP identity boundaries, you must use this account primarily to configure corporate identity infrastructure.
====
```

|Extract service account tokens
|Use localized automated service tokens safely for external system or pipeline authentication actions.
|`oc create token <service_account_name> -n <namespace>`
|===

For guidance on operating your top level administrative accounts, see the following table: 

.Guidance on operating administrative and privileged accounts
[cols="1,2,2",options="header"]
|===
|Action
|Benefit
|Command

|Manage certificates
|Replace default API server certificates with custom certificates issued by an organizational certificate authority. Create a secret containing the certificate and key. Update the API server configuration to reference the custom certificate secret. Certificate rotation occurs automatically when you update the secret with a new certificate.
|`oc create secret tls <secret_name> --cert=<path/to/cert> --key=<path/to/key> -n openshift-config`

|Monitor administrative accounts
|Review cluster events and administrative activities. Query audit logs for specific user activities. Monitor role binding changes indicating privilege escalation attempts.
|`oc get events --all-namespaces`

|Grant namespace administration tasks

> |Empower namespace administrators to handle project-scoped operations: deploying workloads, managing secrets, applying resource quotas, defining network restrictions, and binding local access.

Reviewer: suggestion to change this to something less marketing sounding
```suggestion
|Allow namespace administrators to handle project-scoped operations: deploying workloads, managing secrets, applying resource quotas, defining network restrictions, and binding local access.
```

|Executed through namespace-scoped `oc` subcommands.

|Manage resource quotas
|Configure hard infrastructure resource quotas limiting localized namespace consumption across CPU, memory, storage requests, and PVC limits.
|`oc create quota <quota_name> --hard=cpu=10,memory=20Gi,persistentvolumeclaims=5 -n <namespace>`

|View quota usage
|Audit resource allocation to ensure that project bounds are functioning correctly and limits are not exhausted.
|`oc describe quota -n <namespace>`

|Configure network policies

> |Cluster administrators define cluster-wide network security rules, while namespace administrators create local `NetworkPolicy` objects. Implement default-deny frameworks blocking all unapproved traffic.

Reviewer: do cluster admins exists within managed services? like is that something users can configure? i was under the impression that you can only have `dedicated-admin` privileges bc cluster admins can alter the control plane namespaces, which you cannot do in managed services from what i understand

Reviewer 2: Both roles exist in ROSA: 

<img width="578" height="536" alt="Image" src="https://github.com/user-attachments/assets/4d3e4b83-afe8-4c8f-8c41-4a6fc36eba59" />

Author: Thanks! So, I added a Note at the beginning, as Claude suggested. I actually think it could help understand and use the table. Let me know what you think. I also made changes as you pointed out with the cluster admin related content, nice catch! 

@andykrohg would you please also look at these content changes to ensure technical accuracy?

|`oc create -f <networkpolicy.yaml> -n <namespace>`

|Assign security context constraints

> |Cluster admins manage global security context constraints (SCCs) (such as the default `restricted-v2` or the infrastructure-only `privileged` SCC). Privileged namespace administrators with enough rights assign allowed SCCs directly to local service accounts.

Reviewer: similar question here

Author: Leaving these comments to make it easier to see where I made more significant content changes, as suggested by Doc review.

|`oc adm policy add-scc-to-user <scc_name> -z <service_account_name> -n <namespace>`

|View SCC assignments
|Verify that global or namespace SCC privileges match accurately to the correct infrastructure components and local service accounts.
a|
* Global view: `oc get scc`
* Namespace verification: `oc get scc -o yaml \| grep -A 5 <service_account_name>`
|===

For guidance on decommissioning your top level administrative accounts, see the following table:

.Guidance on decommissioning your administrative and privileged accounts
[cols="1,2,2",options="header"]
|===
|Action
|Benefit
|Command

|Remove user access
|Remove broad administrative or localized namespace role assignments from departing personnel to prevent residual access.
a|

> * Cluster removal: `oc adm policy remove-cluster-role-from-user cluster-admin <username>`

Reviewer: similar question

a|
* Remove User object: `oc delete user <username>`
* Or remove Identity link: `oc delete identity <identity_provider_name>:<username>`

|Remove service account access
|Permanently delete namespace-scoped automated service accounts that are no longer required for active operations.
|`oc delete serviceaccount <service_account_name> -n <namespace>`

|Verify service account deletion
|Auditing step ensures no unlinked automated keys remain active, which effectively blocks associated pods from authenticating unauthorized API calls.
|Monitored by querying namespace resources post-deletion.

|Transfer resource ownership
|Audit resources created by the departing administrator to transfer ownership. Review projects, security policies, and configurations requiring ownership reassignment to active administrators.
|Managed through resource metadata modification and project reassignment.
|===

For guidance on security related settings, see the following table:

.Guidance on setting security for your administrative and privileged accounts
[cols="1,2,2",options="header"]
|===
|Action
|Benefit
|Command

> |Install compliance operator

Reviewer: should this be capitalized because it's about a specific operator?
```suggestion
|Install the Compliance Operator
```

> | Automate configuration auditing and compliance monitoring. The Compliance Operator scans cluster infrastructure against National Institute of Standards and Technology (NIST) 800-53 controls required for FedRAMP. Create `ComplianceSuite` resources to schedule assessments that detect configuration drift and generate remediation recommendations. Visualize compliance results in the OpenShift console or integrate with Red Hat Advanced Cluster Security (ACS) dashboard for centralized compliance tracking and correlation with security findings.

Reviewer: not sure if RHACS also has an attribute
```suggestion
| Automate configuration auditing and compliance monitoring. The Compliance Operator scans cluster infrastructure against National Institute of Standards and Technology (NIST) 800-53 controls required for FedRAMP. Create `ComplianceSuite` resources to schedule assessments that detect configuration drift and generate remediation recommendations. Visualize compliance results in the {OCP-short} console or integrate with Red Hat Advanced Cluster Security (RHACS) dashboard for centralized compliance tracking and correlation with security findings.
```

Author: I didn't find one for RHACS

> |Handled through OperatorHub, then create `ComplianceSuite` resources targeting FedRAMP profiles, `ocp4-moderate` or `ocp4-high`. View results in OpenShift console under `Compliance` or in ACS dashboard.

Reviewer: ```suggestion
|Handled through OperatorHub, then create `ComplianceSuite` resources targeting FedRAMP profiles, `ocp4-moderate` or `ocp4-high`. View results in {OCP-short} console under `Compliance` or in ACS dashboard.
```

|`oc create secret generic <secret_name> --from-file=<path/to/file> -n <namespace>`

|Pull image secrets
|Safely store and configure private registry authentication tokens by creating dedicated `docker-registry` credentials within the project.
|`oc create secret docker-registry <secret_name> --docker-server=<registry-url> --docker-username=<username> --docker-password=<password>`

|Link secrets to service accounts
|Inject private registry pull credentials directly into namespace service accounts to allow automated image validation and pull authentication.
|`oc secrets link <service_account_name> <secret_name> --for=pull -n <namespace>`

|Manage monitoring and logging access
|Namespace administrators access localized application logs and metrics within assigned boundaries, preventing cross-namespace data visibility.
|`oc logs <pod-name> -n <namespace>`

|Review installed operators
|Inspect local, namespace-scoped operators running within the project from OperatorHub, verifying their management permissions are constrained strictly to that namespace.
|`oc get csv -n <namespace>`

|Configure Security Context Constraints (SCCs)

> |Control pod security capabilities. Cluster administrators manage SCCs defining allowable pod security configurations. The default `restricted-v2` SCC applies to pods that do not specify an explicit SCC. The `privileged` SCC grants containers host-level access and should be restricted to infrastructure components only.

Reviewer: similar question

|`oc get scc`

|Define network policies

> |Cluster administrators define cluster-wide network security policies. `NetworkPolicy` resources control pod-to-pod communication and external access. Create default-deny policies requiring explicit allow rules for permitted traffic.

Reviewer: similar question

// Module included in the following assemblies:
//
// * rosa_govcloud/rosa-fedramp-security-config.adoc

:_mod-docs-content-type: REFERENCE

[id="rosa-fedramp-accounts-config_{context}"]
= Red{nbsp}Hat {OCP} FedRAMP Rev5 secure configuration guidance for top-level administrative and privileged accounts

[role="_abstract"]
{product-title} gives you comprehensive frameworks to find roles, isolation boundaries, and operational functions for top-level administrative or privileged accounts. These frameworks apply within a Red{nbsp}Hat {OCP} Federal Risk and Authorization Management Program (FedRAMP) Rev5 Secure Configuration Guidance (SCG) environment.

The guidance you receive addresses the SCG-CSO-RSC (Cloud Service Offering - Recommended Secure Configuration) requirement, including both cluster-wide top-level administrative accounts and namespace-scoped privileged accounts. The parameters in both administrative and privilege accounts help you deploy, operate, decommission, and secure access among your configurations.

> For the most up-to-date account features and structural prerequisites, review the official Red{nbsp}Hat {OCP-short} documentation page for your active deployment service. Your browser must have JavaScript enabled to access and interact with the documentation portal for link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.22[Red Hat OpenShift Container Platform]. 

Author: The build breaks when I apply this feedback, so I'm ignoring it, lol

[id="rosa-fedramp-accounts-config_{context}"]
= Red{nbsp}Hat {OCP} FedRAMP Rev5 secure configuration guidance for top-level administrative and privileged accounts

[role="_abstract"]
{product-title} gives you comprehensive frameworks to find roles, isolation boundaries, and operational functions for top-level administrative or privileged accounts. These frameworks apply within a Red{nbsp}Hat {OCP} Federal Risk and Authorization Management Program (FedRAMP) Rev5 Secure Configuration Guidance (SCG) environment.

The guidance you receive addresses the SCG-CSO-RSC (Cloud Service Offering - Recommended Secure Configuration) requirement, including both cluster-wide top-level administrative accounts and namespace-scoped privileged accounts. The parameters in both administrative and privilege accounts help you deploy, operate, decommission, and secure access among your configurations.

For the most up-to-date account features and structural prerequisites, review the official Red{nbsp}Hat {OCP-short} documentation page for your active deployment service. Your browser must have JavaScript enabled to access and interact with the {OCP-short} documentation portal.

*IMPORTANT:*

> * To access the {cluster-manager-short} infrastructure for the first time, use the temporary, local `kubeadmin` account generated during the initial installation phase.

Reviewer 2: kubeadmin isn't used in ROSA. You can create an admin user in OCM with `rosa create admin` instead

* These root credentials reside securely in the `auth/kubeadmin-password ` file within your local installation directory.
* To comply with FedRAMP identity boundaries, you must use this account primarily to configure corporate identity infrastructure.

For guidance and context on different actions you can take in your top level administrative account, see the following table:

.Actions and benefits for account provisioning and access setup 
[cols="1,2,2",options="header"]
|===
|Action
|Benefit
|Command

|Configure identity providers
|Establish enterprise authentication by configuring an OAuth identity provider. OpenShift supports LDAP, Active Directory, OpenID Connect, SAML, GitHub, GitLab, Google, and HTPasswd providers.

> |`oc edit oauth cluster`

Reviewer 2: OAuth object is not configurable in ROSA - changes get reverted by a webhook. Adding OIDC happens in Hybrid Cloud Console instead

Author: As a general note, when you call out OAuth and Audit logging, I removed these instances. From your comments, it seems like that they're not applicable, and these tables are long enough. However, if you wanted me to keep the entry but provide a different command, or description, just let me know.

|`oc create role <role_name> --verb=<get,list,watch> --resource=<pods,services> -n <namespace>`

|Bind custom roles to namespace users
|Bind custom localized roles to users to guarantee that namespace-scoped personnel only have targeted privileges inside that specific boundary.
|`oc adm policy add-role-to-user <role_name> <username> -n <namespace>`

|Grant service account privileges (Namespace)
|Configure service accounts with localized, privileged permissions for automated application workflows, ensuring minimum access needed for operation.
a|
* Create service account: `oc create sa <service_account_name> -n <namespace>`
* Or grant role: `oc adm policy add-role-to-user <role> -z <service_account_name> -n <namespace>`

|Extract service account tokens
|Use localized automated service tokens safely for external system or pipeline authentication actions.
|`oc create token <service_account_name> -n <namespace>`
|===

For guidance on operating your top level administrative accounts, see the following table: 

.Guidance on operating administrative and privileged accounts
[cols="1,2,2",options="header"]
|===
|Action
|Benefit
|Command

|Organize audit logging

> |Configure audit logging to capture administrative actions. Edit the API server configuration to set the audit policy profile. Available profiles include `Default`, `WriteRequestBodies`, `AllRequestBodies`, and `None`. The `WriteRequestBodies` profile captures request metadata and request bodies for create/update operations. Forward audit logs to external systems by configuring log forwarding through the `ClusterLogForwarder` resource in the `openshift-logging` namespace.

Reviewer 2: Audit logging, to my memory, requires configuration changes to the control plane nodes and will likely be overwritten in ROSA

a|
* Remove User object: `oc delete user <username>`
* Or remove Identity link: `oc delete identity <identity_provider_name>:<username>`

|Remove service account access
|Permanently delete namespace-scoped automated service accounts that are no longer required for active operations.
|`oc delete serviceaccount <service_account_name> -n <namespace>`

|Verify service account deletion
|Auditing step ensures no unlinked automated keys remain active, which effectively blocks associated pods from authenticating unauthorized API calls.
|Monitored by querying namespace resources post-deletion.

|Transfer resource ownership
|Audit resources created by the departing administrator to transfer ownership. Review projects, security policies, and configurations requiring ownership reassignment to active administrators.
|Managed via resource metadata modification and project reassignment.
|===

For guidance on security related settings, see the following table:

.Guidance on setting security for your administrative and privileged accounts
[cols="1,2,2",options="header"]
|===
|Action
|Benefit
|Command

|Install compliance operator

> |Automate configuration auditing. The Compliance Operator scans cluster infrastructure against security benchmarks including National Institute of Standards and Technology (NIST) 800-53 controls. Use `ComplianceSuite` resources to schedule routine assessments.

Reviewer 2: We could supply more detail here, and potentially suggest leveraging ACS dashboard for compliance visualization

Author: Let me know what you think of me rewrite, thanks!
